internal/lsp/cache.go

```diff
@@ -196,17 +195,8 @@ func (c *Cache) Delete(uri string) {
 	c.diagnosticsParseMu.Unlock()
 }
 
-func updateCacheForURIFromDisk(cache *Cache, uri string) (string, error) {
-	parsedURI, err := url.Parse(uri)
-	if err != nil {
-		return "", fmt.Errorf("failed to parse URI: %w", err)
-	}
-
-	if parsedURI.Scheme != "file" {
-		return "", fmt.Errorf("only file:// URIs are supported, got %q", parsedURI.String())
-	}
-
-	content, err := os.ReadFile(parsedURI.Path)
+func updateCacheForURIFromDisk(cache *Cache, uri, path string) (string, error) {
```

Comment:
updateCacheForURIFromDisk has been spared knowledge of the URI path translations and so the path must now be provided by the caller.